stdlib/messaging/activemq-artemis/src/main/java/org/ballerinalang/messaging/artemis/ArtemisResourceValidator.java

```diff
@@ -0,0 +1,159 @@
+            case ArtemisConstants.ON_MESSAGE:
+                validateOnMessageResource(resource, dlog, onErrorAvailable);
+                break;
+            case ArtemisConstants.ON_ERROR:
+                validateOnErrorResource(resource, dlog);
+                break;
+            default:
+                dlog.logDiagnostic(Diagnostic.Kind.ERROR, resource.pos,
+                        "Invalid resource name " + resource.getName().getValue() + " in service, only " +
+                                ArtemisConstants.ON_MESSAGE + " and " + ArtemisConstants.ON_ERROR + " are allowed");
+        }
+    }
+
+    private static void validateOnErrorResource(BLangFunction resource, DiagnosticLog dlog) {
+        List<BLangSimpleVariable> paramDetails = resource.getParameters();
+        if (validateParamDetailsSize(paramDetails, 2, resource, dlog)) {
+            if (!ArtemisConstants.MESSAGE_OBJ_FULL_NAME.equals(paramDetails.get(0).type.toString())) {
+                dlog.logDiagnostic(Diagnostic.Kind.ERROR, resource.pos, INVALID_RESOURCE_SIGNATURE_FOR
+                        + resource.getName().getValue() +
+                        " resource: The first parameter should be an artemis:Message");
+            }
+            if (!"error".equals(paramDetails.get(1).type.toString())) {
+                dlog.logDiagnostic(Diagnostic.Kind.ERROR, resource.pos, String.format(
+                        "Invalid resource signature for %s resource in service : The second parameter should be " +
+                                "artemis:ArtemisError",
+                        resource.getName().getValue()));
+            }
+        }
+    }
+
+    private static void validateOnMessageResource(BLangFunction resource, DiagnosticLog dlog,
+                                                  boolean onErrorAvailable) {
+        List<BLangSimpleVariable> paramDetails = resource.getParameters();
+        if (validateParamDetailsSize(paramDetails, 1, 2, resource, dlog)) {
+            if (!ArtemisConstants.MESSAGE_OBJ_FULL_NAME.equals(paramDetails.get(0).type.toString())) {
+                dlog.logDiagnostic(Diagnostic.Kind.ERROR, resource.pos, INVALID_RESOURCE_SIGNATURE_FOR
+                        + resource.getName().getValue() +
+                        " resource: The first parameter should be an artemis:Message");
+            }
+            if (paramDetails.size() == 2) {
+                validateSecondParam(resource, paramDetails, dlog, onErrorAvailable);
+            }
+        }
+    }
+
+    private static void validateSecondParam(BLangFunction resource, List<BLangSimpleVariable> paramDetails,
+                                            DiagnosticLog dlog, boolean onErrorAvailable) {
+        if (!onErrorAvailable) {
+            dlog.logDiagnostic(Diagnostic.Kind.ERROR, resource.pos, "onError resource " +
+                    "function is not found; any errors that occur due to data binding will be discarded along " +
```

Comment:
Do we need that latter sentence "any errors ..." since we are compiler failing this service?

Reply:
Resolved